src/metabase/api/database.clj

```diff
@@ -765,6 +766,7 @@
                                           (sync.schedules/default-randomized-schedule)))
                                        (when (some? auto_run_queries)
                                          {:auto_run_queries auto_run_queries})))))
+        (future (sync/sync-database! <>))
```

Comment:
hmm, what if we want not to do a full scan?

Reply:
Well, we currently always do a full scan when adding a database. But it's a good point. `sync/sync-database!` could actually look at the `is_full_sync` property of a database and leave the decision whether to scan or not to scan out of the api namespace.

Reply:
um, I don't know the full history about the `is_full_sync` parameter on whether or not it's used to control full sync on database creation, so would be great if you could gather some information about that.

Also, I wonder how do we test stuffs with `future`, I'm looking for a test about "call this API and a sync is triggered"